### PR TITLE
Fix texture samplers to not being last in the property list

### LINKED ISSUE #75454: Shader Sampler2D Uniforms are always listed last in the Inspector
### Godot version

4.0 stable

### System information

Windows 10 960m

### Issue description

When creating sampler2d uniforms in a shader, they get listed at the bottom of the uniform group in the inspector. It is often useful to have them listed with related uniforms, so they should ideally follow the same order as listed in the shader code. For instance, albedo texture followed by the albedo color.

This example shows the standard 3d material converted to a shader with some small edits for readability. Note that the 3 sampler 2d uniforms are displayed at the bottom in the inspector and not in the shader order. However, they do display in the relative order that they appear in the shader code.

sampler2d_order

@Chaosus thanks for your fix on the alphabetical uniform group ordering! I thought it would be a good time to add this issue as the final obstacle to getting the perfect uniform ordering. Pretty sure this same issue also happened in 3.x and is a long standing problem.

### Steps to reproduce

- create a standard 3d material
- convert this material to a shader
- note that all of the uniform sample2d are listed at the bottom in the outliner

### Minimal reproduction project

[sampler2d_order.zip](https://github.com/godotengine/godot/files/11098493/sampler2d_order.zip)

### PR DESCRIPTION
**Pull request by Chaosus**
- Fix https://github.com/godotengine/godot/issues/75454

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/88d9325065a3e00e9e168ffad4ff93e12455a357), it works as expected.

Example with a new StandardMaterial3D converted to ShaderMaterial:

Before | After
-|-
image | image

File: servers/rendering/shader_language.h [hunk 1]
```diff
@@ -700,6 +701,10 @@ class ShaderLanguage {
 			String group;
 			String subgroup;
 
+			_FORCE_INLINE_ bool is_texture() const {
+				return order < 0;
+			}
```
Comment (Calinou) [inline servers/rendering/shader_language.h:706]: I'd add a comment to explain why this works somehow, as it's not obvious at a glance.

Comment (Chaosus) [reply servers/rendering/shader_language.h]: Done

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**